### ISSUE TITLE
[Feature] Implement padding for tensordicts

### ISSUE THREAD

**Issue by vmoens** [facebookresearch/rl#246]
TensorDict is the TorchRL data carrier class.
All tensors of a TensorDict share the same batch size, which is defined upon instantiation:
```python
td = TensorDict({"a": torch.ones(3, 4, 5), "b": torch.zeros(3, 4, 1, 3)}, batch_size=[3, 4])
```
In this example, `batch_size` could also be `[3]` or `[]`.

We'd like to be able to pad all the tensors in a tensordict to match a desired batch size. A `pad` function should execute:
```python
>>> dim0_left, dim0_right, dim1_left, dim1_right = [0, 1, 0, 2]
>>> td = TensorDict({"a": torch.ones(3, 4, 1), "b": torch.zeros(3, 4, 1, 1)}, batch_size=[3, 4])
>>> td_pad = pad(td, [dim0_left, dim0_right, dim1_left, dim1_right], value=0.0)
>>> print(td_pad)
TensorDict({
    "a": Tensor(4, 6, 1),
    "b": Tensor(4, 6, 1, 1)}, batch_size=[4, 6])
```
with the zeros appropriately set.
